website/blog/2024-03-21-using-your-private-git-repository-context-for-completion/index.md

```diff
@@ -0,0 +1,113 @@
+---
+authors: [icycodes]
+tags: [deployment, repository context]
+---
+
+# Connect Private GitHub Repository to Tabby
+
+A few months back, we published a blog [Repository context for LLM assisted code completion](https://tabby.tabbyml.com/blog/2023/10/16/repository-context-for-code-completion), introducing the `Repository Context` feature in Tabby.
+
+In this blog, I will guide you through the process of setting up a Tabby server configured with **a private Git repositories** context.
+
+## Generating a Personal Access Token
+
+In order to provide the Tabby server with access to your private Git repositories, it is essential to create a Personal Access Token (PAT) specific to your Git provider. The following steps outline the process with GitHub as a reference:
+
+1. Visit [GitHub Personal Access Tokens Settings](https://github.com/settings/tokens?type=beta) and select `Generate new token`.
+   ![GitHub PAT Generate New Token](./github-pat-generate-new-token.png)
+2. Enter the `Token name`, specify an `Expiration` date, an optional `Description`, and select the repositories you wish to grant access to.
+   ![GitHub PAT Filling Info](./github-pat-filling-info.png)
+3. Within the `Permissions` section, ensure that `Contents` is configured for `Read-only` access.
+   ![GitHub PAT Contents Access](./github-pat-contents-access.png)
+4. Click `Generate token` to generate the new PAT. Remember to make a copy of the PAT before closing the webpage.
+   ![GitHub PAT Generate Token](./github-pat-generate-token.png)
+
+For additional information, please consult the documentation on [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+**Note**: For users of GitLab, guidance on creating a personal access token can be found in the documentation [Personal access tokens - GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token).
+
+## Configuration
+
+To configure the Tabby server with your private Git repositories, you need to provide the required settings in a YAML file. Create and edit a configuration file located at `~/.tabby/config.yaml`:
+
+```yaml
+## Add the private repository
+[[repositories]]
+name = "my_private_project"
+git_url = "https://<PAT>@github.com/icycodes/my_private_project.git"
+
+## More repositories can be added like this
+[[repositories]]
+name = "another_project"
+git_url = "https://<PAT>@github.com/icycodes/another_project.git"
+```
+
+For more detailed about the configuration file, you can refer to the [configuration documentation](https://tabby.tabbyml.com/docs/configuration).
+
+**Note:** The URL format for GitLab repositories may vary, you can check the [official documentation](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#clone-repository-using-personal-access-token) for specific guidelines.
+
+## Building the Index
+
```

Comment:
Add something here to briefly talk about how we are building the index - for example, tree-sitter is used to parse your code and extract critical code components for index building. The index is for fast retrieval (related code snippets) during code completion time to enhance code generation context. Something like this would be helpful.

Reply:
Updated in [08ade79](https://github.com/TabbyML/tabby/pull/1705/commits/08ade79e47c2db2bf37431b711e34765a0ac3e97)